### PR TITLE
move build storage where it is owned

### PR DESCRIPTION
**Pull request by deads2k**
This moves the build REST storage into the build package for clean separation, maintenance, and composition.

@bparees @csrwng

### CODE REVIEW

File: pkg/build/apiserver/apiserver.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+	buildlogregistry "github.com/openshift/origin/pkg/build/registry/buildlog"
+	"github.com/openshift/origin/pkg/build/webhook"
+	"github.com/openshift/origin/pkg/build/webhook/bitbucket"
+	"github.com/openshift/origin/pkg/build/webhook/generic"
+	"github.com/openshift/origin/pkg/build/webhook/github"
+	"github.com/openshift/origin/pkg/build/webhook/gitlab"
+	imageclient "github.com/openshift/origin/pkg/image/generated/internalclientset"
+)
+
+type BuildConfig struct {
```
Comment (csrwng) [inline pkg/build/apiserver/apiserver.go]: s/BuildConfig/BuildServerConfig/ 
Just to avoid confusion

**Comment by deads2k** [comment]
> Is the thinking that this will eventually be run as a separate server?

Someday, yes.  This positions us for that along with making ownership of the storage clearer and seeing the "weird" dependencies like: "OMG, the build apiserver is directly calling the kubelet so we can never run this impl on kubernetes". 

Speaking of which, we need to fix that.

File: pkg/build/apiserver/apiserver.go (hunk 1 above)
Comment (deads2k) [reply pkg/build/apiserver/apiserver.go]: > s/BuildConfig/BuildServerConfig/
> Just to avoid confusion

done

**Comment by deads2k** [comment]
Comments addressed, applying label per comment.

**Pull request merged**